Sources/NIOHTTPCompression/HTTPCompression.swift

```diff
@@ -0,0 +1,156 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CNIOExtrasZlib
+import NIO
+
+public enum NIOHTTPCompressionSettings {
```

Comment:
@adam-fowler / @Lukasa / @glbrntt so I proposed this `NIOHTTPCompressionSettings` as the namespace but then there are also errors in there so 'Settings' isn't that great.

Do we have better ideas that satisfy the following constraints:

- start with `NIO*`
- are not `NIOTHTTPCompression` (as that's the module name)

Reply:
We could pull `NIOCompression` one and then call the nested ones just `public struct Algorithm` and `public struct Error`. So `NIOCompression.Algorithm`.

It's not perfect because obviously we may later want to compress stuff that's not HTTP but at least it reads nicely ;)

Reply:
WFM.

Reply:
renamed

Reply:
Given the concern over possible later use of compression in NIO. Should the ByteBuffer compression be separated out from the the NIOHTTPCompression module and made for more general usage?